feat!: add landing page prefooter component

christian-dinh opened the pull request:
## Overview

### PR Checklist

- [x] Related to designs: [yerp](https://www.figma.com/file/2aBlW1pgFt5PT6soRu1HXy/%E2%9C%8F%EF%B8%8F-Landing-Page-Templates---Unification?node-id=1%3A43)
- [x] Related to JIRA ticket: [REACH-409](https://codecademy.atlassian.net/browse/REACH-409)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

contains the following refactors:
- all `LandingPage*` exports have been shortened to `Page*` [BREAKING]
- all Landing Page components in `labs/brand` have been moved to `labs/landingPage`, as have their stories
- all Landing Page file names are now unprefixed

![Untitled](https://user-images.githubusercontent.com/19958143/100024100-2ac74e00-2db4-11eb-97e7-397702a69e10.gif)

JoshuaKGoldberg commented on packages/gamut-labs/src/landingPage/Prefooter.tsx:
```diff
@@ -0,0 +1,53 @@
+export type PagePrefooterProps = {
+  title?: string;
+  desc?: string;
+  cta?: string;
+  ctaHref?: string;
```
Is it intended in the API for this component that if you provide a `cta`, you must provide a `ctaHref` (and vice versa)? If so, you might consider doing something fancy with the type definition here to enforce it at the type system level.

One option would be to make it a nested object, like `cta: { children: React.ReactNode; href: string; }`. 

Another might be something like `type PagePrefooterProps = PagePrefooterPropsWithCta | PagePrefooterPropsWithoutCta`, where the `*WithCta` version has both as required properties.

Non-blocking & up to you 😄

christian-dinh replied:
didn't know that was an option, yes that sounds like exactly what I want here ty!

JoshuaKGoldberg commented on packages/gamut-labs/src/index.ts:
```diff
@@ -1,3 +1,4 @@
 export * from './brand';
 export * from './experimental';
+export * from './landingPage';
```
Should this be `./page`?

![Screenshot of Labs > LandingPage in the storybook preview](https://user-images.githubusercontent.com/3335181/100084302-993df780-2e18-11eb-94de-5ad920036102.png)

christian-dinh replied:
I'd actually like some opinions on this, this was intentional but I'm not sure if it's normal to have a distinction between story directory and export prefix

JoshuaKGoldberg commented on packages/gamut-labs/src/landingPage/CTA.tsx:
```diff
@@ -7,13 +7,13 @@ const CTA = styled.div`
 `;
 
-export type LandingPageSectionCTAProps = {
+export type PageSectionCTAProps = {
```
My two cents is still that I prefer "marketing"... "page" is kind of a generic term that could apply to other things in Gamut. Not a blocker if you really prefer page, but definitely a preference.

JoshuaKGoldberg reviewed (approved):
Cool!

Some organizational work to be done, maybe (see screenshot), but generally super. If you intended this to be under LandingPage intentionally in Storybook then ignore me!

rogervera reviewed (approved):
LGTM! I'm for `page` and internally in Reach we formed a consensus on using that for the prefix. Josh does bring an outsider perspective on `marketing` perhaps being more clear on these components' intent at the cost of a `few` extra chars. We can decide in standup.

+1 on the type definition suggestions as well

christian-dinh commented:
some thoughts re: `Page` vs `Marketing`

LandingPage:
- too specific (these components could be used on other pages)
- too long 😩 

Page:
- possibly too broad (literally every react component ends up on a "page")
- really short 🎉 

Marketing:
- probably the most accurate 🎯 
- medium length

so far all these components are only being imported in one file, so I'm not sure how much it all matters either way 🤷‍♀️

I'm personally cool with `Page` but naming is hard and this does seem like something we might revisit if these components start getting used for other things

codecademydev commented:
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@6.0.0-alpha.afb22e.0<br/>@codecademy/styleguide@19.0.0-alpha.afb22e.0<br/></pre><!--LERNA_VERSION_COMMENT-->

christian-dinh commented:
went with
```
cta?: {
  text: string;
  href: string;
}
```
for the cta typing, created a `BaseProps` export in `types.tsx` that all landing page organisms share, and updated the consuming components and stories

messed with storybook a little bit to see if there was a better way to get object props to work in the live editor, but there isn't really. Since there is no way to replace an object arg with `undefined` in the storybook ui, I just added stories to `PageHero` and `PagePrefooter` without CTAs. Looks like this experience might get better soon?

https://github.com/storybookjs/storybook/pull/12824

codecademydev commented:
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbd6de661aaaa3a5162bbb6--gamut-preview.netlify.app>https://5fbd6de661aaaa3a5162bbb6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbd6de661aaaa3a5162bbb6>Deploy Logs</a></p><!--ALPHA_DEPLOY_COMMENT-->
